### PR TITLE
Taking Brute or Burn Damage Reduces Stuns

### PR DESCRIPTION
**Pull request by SkeletalElite**
## About The Pull Request
Regular stuns are reduced by a number of seconds equal to the damage dealt * .35 seconds. Only triggers on more than 5 damage
Things that knock you completely unconscious/asleep are unchanged. 

Here is a [Demo](https://youtu.be/mUELIMhLiKg)

(demo somewhat out of date)

## Why It's Good For The Game
 Stuns have to be balanced around the fact that they are lethal, but at the same time there has to be effective methods in place to non lethally detain players. This creates a precarious situation to balance around because it's very difficult to make a stun good enough for security and non lethal detainment, but not so good it's one of the best ways to kill people.

At the very least, to kill someone off the back of a single stun with this change, you MUST cuff them first. The more steps there are between stunned and dead, the less we have to be concerned that a stun is too good as a way to kill people but also remain an effective tool for non lethal detainment. 

Of course, this doesn't make stuns completely useless in combat, either. A 10-second stun will generally give you about 20 free damage that the person you're attacking can do nothing about. As for stam crit. It will end 2 seconds after taking brute or burn, which is 1 or 2 free hits. Be warned that if the attack that stam crits them also does damage those 2 seconds start instantly though,

I'm considering increasing cuff time as well. The idea is to increase cuff time by 1-second but make black gloves/gripper gloves/gorilla gloves give a 1-second bonus to cuff speed. Cable cuffs would have an additional half second penalty compared to zip ties and handcuffs. However, I'm not going to do that until I see how this PR goes.
## Changelog
:cl:
balance: Taking Brute or Burn damage will reduce stuns by a number of seconds equal to .35x the damage taken. Only triggers if the damage is greater than 5.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -1430,6 +1433,9 @@ GLOBAL_LIST_EMPTY(features_by_species)
 		if(BURN)
 			H.damageoverlaytemp = 20
 			var/damage_amount = forced ? damage : damage * hit_percent * burnmod * H.physiology.burn_mod
+			H.AdjustAllImmobility(damage_amount * -1/2 SECONDS)
```
Comment (san7890) [inline code/modules/mob/living/carbon/human/species.dm:1436]:
```suggestion
			H.AdjustAllImmobility(damage_amount * -0.5 SECONDS)
```

this is WAY better than fractions.

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -1422,6 +1422,9 @@ GLOBAL_LIST_EMPTY(features_by_species)
 		if(BRUTE)
 			H.damageoverlaytemp = 20
 			var/damage_amount = forced ? damage : damage * hit_percent * brutemod * H.physiology.brute_mod
+			H.AdjustAllImmobility(damage_amount * -1/2 SECONDS)
```
Comment (san7890) [inline code/modules/mob/living/carbon/human/species.dm:1425]:
```suggestion
			H.AdjustAllImmobility(damage_amount * -0.5 SECONDS)
```

**Review by san7890 — CHANGES_REQUESTED** [review]
i don't know how i feel about this in practice yet but please do this first

**Comment by blackdav123** [comment]
I like the intent but I can imagine tons of ways this can be abused to make sec unable to arrest people.

Self harm like a chem mix that damages and heals should be accounted for, as something like that shouldnt be a get out of baton free card.

In fights with multiple people, sec would likely be unable to stop riots without resorting to lethals because any assistant that gets batonned twice could just be punched once by an ally to get them back up.

Fire also becomes an issue because any maniac could just light themself on fire and get the same effect. Pair this with something that reduces the damage taken from fire and sec has major problems.

**Comment by san7890** [comment]
@Mothblocks changes to stun combat to require immediately cuffing and other things that might have an affect on the head coder ruling, what are your thoughts?

**Comment by Mothblocks** [comment]
> Nerfing stun batons will make nonlethal arrests from security significantly less desirable. Antagonists have plenty of tools to kill, and while other tools such as disablers, pepper spray, etc have their place, do not have the consistency of a stun baton. An average level skilled player should still be able to reasonably participate in security.

✅ Pass, with an admittedly clever idea

> Stealth antagonists, especially ones that do not have gear (like obsessed), rely on tools like stun batons because they are fairly accessible and good at enabling this playstyle. It is already a common issue that antagonists will play “loud”, where they are not secret about the fact they are an antagonist–stealth should not be further discouraged.

:x: Fail, you have made it less desirable to go loud while buffing going loud. It is now significantly harder for a security officer to kill someone going loud, whilst antagonists who don't go loud now need more gear to do the same thing.

> There is more to fights than individuals vs individuals/teams. Fights with multiple people on both sides are very fun, and stun batons are much more balanced in this environment. Making a change like changing the stuns to require 3 hits would make them completely useless in these fights.

❓ Questionable. If you have multiple people it's still fairly easy to kill someone with this change, but it also makes the fight more dynamic if the other side can rescue the previously stunned person in time.

This is better than the lower effort PRs but I still don't think I'm satisfied. I think you are on the right track, though.

**Comment by GuillaumePrata** [comment]
Late, but I have to say that this is one of the smarter ways I have seen people try to balance stun murderboning, very creative but still simple.

**Comment by SkeletalElite** [comment]
> It is now significantly harder for a security officer to kill someone going loud, whilst antagonists who don't go loud now need more gear to do the same thing.

I don't entirely agree with this.
Security officers who are chasing an antagonist going loud typically do not need to outright kill the antagonist on the spot because they are typically running solo. They can capture them and cuff them before killing them, which also indirectly encourages using execution room/perma over just harm batoning to death on the spot.  Although, the freedom implant may need to be nerfed as a result of this change. 

I can see where you're coming from with stealthy antags needing more gear to get by though, since they will need cable cuffs to capture people if they're relying on a stun baton as their weapon.

**Comment by Mothblocks** [comment]
Hmm okay let's test this

**Comment by nikothedude** [comment]
This PR HAS to account for self-inflicted damage, or potentially unintentional damage

I am dead certain there are some chems out there that continuously do brute/burn damage that are easily outhealable by other chems

Fire/High temp will make you unstunnable as well

**Comment by ATH1909** [comment]
we might need to slightly tweak some things that are supposed to both stun and deal damage.

what's the order of operations for, say, taking damage+a stun from being slammed into a wall? do you take the damage first and then the stun, or do you take the stun first and then get it reduced by the damage?

is rubbershot gonna be way, way harder to use now because accidentally "overkilling" someone with an extra rubbershot blast will cause them to just get up after 2 seconds?

**Comment by SkeletalElite** [comment]
Chem damage already doesn't work

Fire damage/cold does but I'm not certain it's an issue yet

**Comment by Mothblocks** [comment]
> what's the order of operations for, say, taking damage+a stun from being slammed into a wall? do you take the damage first and then the stun, or do you take the stun first and then get it reduced by the damage?

good point

> This PR HAS to account for self-inflicted damage, or potentially unintentional damage

good point

**Comment by SkeletalElite** [comment]
> is rubbershot gonna be way, way harder to use now because accidentally "overkilling" someone with an extra rubbershot blast will cause them to just get up after 2 seconds?

Like I said rubbershot is going to need some changes its pretty much useless with this PR right now.
Wall throw as far as I know doesn't actually do any damage, just stuns. If it does do damage I believe the damage will reduce the stun though

**Comment by nikothedude** [comment]
Theres a ton of edge cases, sa well, thta would probably cause issues

Ex. Viruses, depressurization, a bunch of greytiders running around and slapping people to get them up from stamcrit, flipping (if yo ucan flip while stamcrit, you can on-demend hurt yoruself), pyroxadone + clf4 to keep yourself always on fire but healing, more and more 

I like the idea in concept and think it's on the right track but this is gonna run into a lotta issues

**Comment by Mothblocks** [comment]
> It hooks into apply_damage, which stuff like viruses and chems don't use (primarily used by combat)

Wait wat, why don't they. We shouldn't rely on that

**Comment by Mothblocks** [comment]
We shouldn't rely on that always being the case .-.

**Comment by SkeletalElite** [comment]
we could make it only trigger on >5 damage. I guess you could technically stun someone and beat them with your pocket o2 tank or something but it would take more over 20 hits just to crit them

**Comment by Mothblocks** [comment]
I test merged this on Sybil and will run a poll after a week or so

**Comment by Fikou** [comment]
> > It hooks into apply_damage, which stuff like viruses and chems don't use (primarily used by combat)
> 
> Wait wat, why don't they. We shouldn't rely on that

apply_damage checks stuff like armor, shields etc, its for physical stuff, i think its an ok system

**Comment by Rex9001** [comment]
> I test merged this on Sybil and will run a poll after a week or so

I dont know how hard this is, but can we also test merge this on Terry? Because that is a large part of our playerbase

**Comment by Cheshify** [comment]
This is actually very interesting, excited to see it tested more.

**Comment by SinfulBliss** [comment]
So you can free someone from stamcrit by punching them?

**Comment by saprasam** [comment]
wouldn't this nerf heretic due to the grabby hand being its only viable stun option

**Comment by Mothblocks** [comment]
> I dont know how hard this is, but can we also test merge this on Terry? Because that is a large part of our playerbase

I was trying but I can't actually connect to the box lol

**Comment by SkeletalElite** [comment]
Okay I made it only trigger if an attack does 5 or more damage this ensures very low damage cannot be used to unstun you. Rubbershot will now work fine because it counts each pellet seperately. Makeshift shells do 6 damage each and therefore still trigger.

**Comment by SinfulBliss** [comment]
> wouldn't this nerf heretic due to the grabby hand being its only viable stun option



> Okay I made it only trigger if an attack does more than 5 damage this ensures very low damage cannot be used to unstun you. Rubbershot will now work fine because it counts each pellet seperately. Makeshift shells do 6 damage each and therefore still trigger. Beanbag slugs are still getting cucked by this because they do 10 damage, but I think we could just reduce their damage.

still, can't you just hit someone with a crowbar and free them from stamcrit? also will be a bit brutal for heretics who almost always need to stamcrit a target to kill it properly, and have to do so quickly

**Comment by moocowswag** [comment]
This doesnt actually remove the stun baton meta it just changes it to stun baton cuff meta since you can just kill people after cuffing

But it definitely does nerf stuns in large scale combat.

**Comment by Bawhoppen** [comment]
This encourages lethal sec...

👍 x 💯

**Comment by pilotmans** [comment]
This is just a bad idea in concept

**Comment by itseasytosee** [comment]
Not sure if this is the best balancing lever for the mess that is stamcrit but hey it's something

**Comment by Fritzlschnitzel** [comment]
hecking based

**Comment by ATH1909** [comment]
> > is rubbershot gonna be way, way harder to use now because accidentally "overkilling" someone with an extra rubbershot blast will cause them to just get up after 2 seconds?
> 
> Like I said rubbershot is going to need some changes its pretty much useless with this PR right now. Wall throw as far as I know doesn't actually do any damage, just stuns. If it does do damage I believe the damage will reduce the stun though

borg harmspins do damage when you slam into a wall, IIRC

**Comment by ATH1909** [comment]
> Makeshift shells do 6 damage each and therefore still trigger.

What if your target is wearing armor?

Also, wasn't there a refactor or something to buckshot that made it wound and stuff in one big chunk? or am I misremembering?

**Comment by SkeletalElite** [comment]
> > Makeshift shells do 6 damage each and therefore still trigger.
> 
> What if your target is wearing armor?
> 
> Also, wasn't there a refactor or something to buckshot that made it wound and stuff in one big chunk? or am I misremembering?

No wounding with pellet clouds is a snowflake check I think. It works for wounds but it doesn't apply to this. Currently, if you have armor that is reducing damage to less than 5 it won't trigger, 

Theoretically if you are armored and didn't take more than 5 per pellet it wouldn't unstun you.

**Comment by ATH1909** [comment]
will this cause people to spontaneously get up after being run over by a mulebot

**Comment by LemonInTheDark** [comment]
> > is rubbershot gonna be way, way harder to use now because accidentally "overkilling" someone with an extra rubbershot blast will cause them to just get up after 2 seconds?
> 
> Like I said rubbershot is going to need some changes its pretty much useless with this PR right now. 

You could store when a stun last happen, and ignore any attempted stun resets that happen in the same tick (IE: avoiding the brute + other meme)
That may be a bit too gamey tho, unsure

**Comment by YuliceTheDruid** [comment]
Interesting but I can see this being absolutely horrendous for seccies when someone runs up with a fire extinguisher or an air tank and taps the guy on the ground. Absolutely going to encourage sec to just Kill People, unless that's the intent.

**Comment by TheMightyAltroll** [comment]
Not really a fan of this. It feels like a bandaid on a much bigger problem. Ultimately we need an alternative not-so-obnoxious method of nonlethally subduing people. I recommend delving into upgrading our melee with some ideas from Goon's wrestling/stamina system.

I had written this up based upon my ideas for how to improve combat, have a read if you like: [https://hackmd.io/@TheMightyAltroll/rygFG9SZh](Stun Combat Changes)

File: code/modules/mob/living/carbon/human/species.dm [hunk 3]
```diff
@@ -1422,6 +1422,10 @@ GLOBAL_LIST_EMPTY(features_by_species)
 		if(BRUTE)
 			H.damageoverlaytemp = 20
 			var/damage_amount = forced ? damage : damage * hit_percent * brutemod * H.physiology.brute_mod
+			if(damage_amount > 5)
+				H.AdjustAllImmobility(damage_amount * -1/2 SECONDS)
```
Comment (san7890) [inline code/modules/mob/living/carbon/human/species.dm:1426]: why did you regress this back to the fractions

**Review by san7890 — CHANGES_REQUESTED** [review]
yeah the whole "order of operations" stuff leaves something to be desired as far as logistics of this mechanic working, but as previously mentioned this would be a real shitfest to accomplish in the current damage framework

HOWEVER, aside from the whole potential-to-incentivize-going-loud stuff, if you were able to disentangle or otherwise make it work such that only the initial assailant/assailing item would provoke the ending of the stun - this would be a very strong PR. of course mothblocks would be the final say and all of that, but the common complaint is "it's very easy to cheese" which I am forced to agree with as far as this implementation is concerned.

if it's found to be good and acceptable at the end of this it's whatever but decreasing the cheese factor is a critical key to furthering the game and not having this be chalked up to a mistake in a few month's time. your choice ultimately

**Comment by Mothblocks** [comment]
you can close the pr if you dont like it san. youre the one who assigned it to me

**Comment by san7890** [comment]
yea but you closed it so that's where i got that confusion from

i wanna see if it's feasible for it to dig itself out of this hole of cheese first (may as well since it's already undergoing ongoing testmerges)

**Comment by YuliceTheDruid** [comment]
So uh, this kind of makes Xenomorphs impossible to play as, having just done a round as a xeno in a test merge.

**Comment by LemonInTheDark** [comment]
image

**Comment by Bawhoppen** [comment]
I'm just gonna go ahead and mark your comments as irrelevant: working as designed

File: code/modules/mob/living/carbon/human/species.dm (hunk 3 above)
Comment (SkeletalElite) [reply code/modules/mob/living/carbon/human/species.dm:1426]: That was an accident oops, I'll fix it

**Comment by DrDiasyl** [comment]
I'm just imagining people beating themselves when security with baton approaches lol.

Great PR anyway.

**Comment by SkeletalElite** [comment]
Stam crit is now reduced by half of damage taken like regular stuns.

So if you hit someone with 6 damage weapon you're only reducing stam crit by 3 seconds. I haven't tested how this interacts with pellet clouds yet but I think each individual pellet should apply this effect seperately.

This should help with anti stun cheese as well since a longer stun needs more damage to knock you out of it.

**Comment by SkeletalElite** [comment]
> image

In my testing it takes 2 hits for a xeno to unstun and since they can just right click stun you they can just left click > right click > left click. if xenos cant keep someone down seems like a skill issue

**Comment by Mothblocks** [comment]
don't nerf xenos.

**Comment by SkeletalElite** [comment]
I reduced the amount of stun taken away by damage for non stam crit stuns to .35 x damage. Xenos stun for 10 seconds when they right click. Xeno slashes reduce that by 7 seconds with the most recent change and so you simply have to restun them after slashing them. Which will put their stun back to 10 seconds. I don't think it's untuitive, restunning to keep someone stunned has been like the core of fighting for years if you didn't have enough damage to kill in a single stun.

I realized I was slightly wrong earlier about 1 hit not stunning (it would but you could stun someone again during the standup timer and it required very precise timing) but now, after hitting someone once they will be stunned for 3 seconds still and you can just restun them because xenos have unlimited 10 second stuns.

**Comment by Mothblocks** [comment]
@optimumtact ?

**Comment by optimumtact** [comment]
nobody has any intention of merging this so you are just unfairly stringing the OP along

**Comment by moocowswag** [comment]
If the pr was merged it would have just made the game harder for new players while stuff like this just knowledge gates them anyway.

**Comment by JohnFulpWillard** [comment]
@Mothblocks I still think the poll is a good idea because I actually think this is a clever way of going about it

**Pull request closed without merging**